bears/c_languages/codeclone_detection/ClangComplexityBear.py

```diff
@@ -0,0 +1,98 @@
+from clang.cindex import Index, CursorKind
+
+from coalib.bears.GlobalBear import GlobalBear
+from bears.c_languages.ClangBear import clang_available
+from coalib.results.Result import Result
+
+
+class ClangComplexityBear(GlobalBear):
+    """
+    Calculates cyclomatic complexity of each function and displays it to the
+    user.
+    """
+    check_prerequisites = classmethod(clang_available)
+    decisive_cursor_kinds = {
+        CursorKind.IF_STMT, CursorKind.WHILE_STMT, CursorKind.FOR_STMT
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(ClangComplexityBear, self).__init__(*args, **kwargs)
+        self.exit_points = 0
+        self.decision_points = 0
+        self.complexities = {}
```

Comment:
It will be much pain to pass them around, but I agree that their initialization should go into run